sn_interface/src/network_knowledge/section_authority_provider.rs

```diff
@@ -307,6 +306,32 @@ pub mod test_utils {
             (section_auth, nodes)
         }
 
+        /// Generate a `SectionAuthorityProvider` for testing with the provided set of members.
+        pub fn random_sap_with_members<R: RngCore>(
+            rng: &mut R,
```

Comment:
`RngCore` is implemented for `&mut RngCore`. The `rand` people did this so that we don't have to request mutable references to the rng's:

So you can rewrite this as:
```suggestion
            mut rng: impl RngCore,
```

Any function that takes a &mut RngCore can be rewritten to this